kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/Range.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider;
+
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Represents the set of integers between two integer endpoints. So the Range
+ * with startInclusive 1 and endExclusive 3 contains the integers 1 and 2.
+ *
+ * A Range must be non-empty, endExclusive must be greater than startInclusive
+ */
+class Range {
```

Comment:
could this be a record?

Reply:
yes, I started out extending `AbstractSet` and backed off tying it to the Collections api. It should become a record now.